### PR TITLE
tests: benchmarks:  Verify wakeup for NFC from s2ram for nrf54h20

### PR DESCRIPTION
**Pull request by nordic-bami**
Verify the NFC with s2ram for nrf54h20

### CODE REVIEW

File: tests/benchmarks/current_consumption/nfc_idle/Kconfig.sysbuild [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
```
Comment (nordic-segl) [inline tests/benchmarks/current_consumption/nfc_idle/Kconfig.sysbuild:2]: :smile:

**Review by nordic-segl — APPROVED** [review]

**Review by nordic-piks — APPROVED** [review]

File: tests/benchmarks/current_consumption/nfc_idle/sysbuild.cmake [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
```
Comment (nordic-segl) [inline tests/benchmarks/current_consumption/nfc_idle/sysbuild.cmake:2]: :smile:

**Review by nordic-segl — APPROVED** [review]

File: tests/benchmarks/current_consumption/nfc_idle/sysbuild.cmake [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+if(SB_CONFIG_SOC_NRF54H20)
+  if("${SB_CONFIG_REMOTE_BOARD}" STREQUAL "")
+    message(FATAL_ERROR "REMOTE_BOARD must be set to a valid board name")
+  endif()
+
+  # Add remote project
+  ExternalZephyrProject_Add(
+      APPLICATION remote
+      SOURCE_DIR ${ZEPHYR_BASE}/../nrf/tests/benchmarks/power_consumption/common/remote_sleep_forever
```
Comment (tejlmand) [inline tests/benchmarks/current_consumption/nfc_idle/sysbuild.cmake:15]: Please use `SYSBUILD_NRF_MODULE_DIR` instead.
```suggestion
      SOURCE_DIR ${SYSBUILD_NRF_MODULE_DIR}/tests/benchmarks/power_consumption/common/remote_sleep_forever
```

File: tests/benchmarks/current_consumption/nfc_idle/Kconfig.sysbuild [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+source "${ZEPHYR_BASE}/share/sysbuild/Kconfig"
+
+config REMOTE_BOARD
+	string "The board used for remote target"
```
Comment (tejlmand) [inline tests/benchmarks/current_consumption/nfc_idle/Kconfig.sysbuild:10]: afaict there is no reason for this field to have a prompt.

It's much cleaner (and simpler) to adjust the board qualifier based on the SoC.
This further has the benefit of supporting all boards based on this SoC and not having to create a dedicated sysbuild conf file for each board using the SoC.

Like this:
```suggestion
	string
	default "$(BOARD)/nrf54h20/cpurad" if SOC_NRF54H20_CPUAPP 
```

File: tests/benchmarks/current_consumption/nfc_idle/sysbuild.cmake [hunk 5]
```diff
@@ -0,0 +1,24 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+if(SB_CONFIG_SOC_NRF54H20)
+  if("${SB_CONFIG_REMOTE_BOARD}" STREQUAL "")
+    message(FATAL_ERROR "REMOTE_BOARD must be set to a valid board name")
+  endif()
```
Comment (tejlmand) [inline tests/benchmarks/current_consumption/nfc_idle/sysbuild.cmake:10]: with the proposed changes in `Kconfig.sysbuild`, then as soon the nRF54h20 is used then the remote board is always set, so in principle this check can be removed.

But as it does no harm I leave the decision to the author.

File: tests/benchmarks/current_consumption/nfc_idle/sysbuild/nrf54h20dk_nrf54h20_cpurad.conf
Comment (tejlmand) [inline tests/benchmarks/current_consumption/nfc_idle/sysbuild/nrf54h20dk_nrf54h20_cpurad.conf:1]: please remove this file as it is no longer needed.

File: tests/benchmarks/current_consumption/nfc_idle/testcase.yaml [hunk 6]
```diff
@@ -12,5 +12,24 @@ tests:
...
+      - nrf54h20dk/nrf54h20/cpuapp
+    integration_platforms:
+      - nrf54h20dk/nrf54h20/cpuapp
+    extra_args:
+      - CONFIG_PM=y
+      - CONFIG_PM_S2RAM=y
+      - CONFIG_PM_S2RAM_CUSTOM_MARKING=y
+      - CONFIG_BOOT_BANNER=n
+      - CONFIG_SOC_NRF54H20_NO_MRAM_LATENCY=n
+      - SB_CONF_FILE=sysbuild/nrf54h20dk_nrf54h20_cpurad.conf
```
Comment (tejlmand) [inline tests/benchmarks/current_consumption/nfc_idle/testcase.yaml:29]: Not needed.
```suggestion
```

**Review by tejlmand — APPROVED** [review]

**Pull request merged**